APR-Core/Party.lua

```diff
@@ -265,3 +265,57 @@ function APR.party:UpdateGroupListing(steps, username)
     end
     UpdateGroupStep()
 end
+
+function APR.party:CheckIfPartyMemberIsFriend()
+
+    -- If Player is NOT in a Group (not instance), do nothing
+    if not IsInGroup("LE_PARTY_CATEGORY_HOME") then
+        return false
+    end
+
+    -- Get FriendLists (WoW and BNet)
+    local FriendListTable = GetFriendsList()
+
+    -- Check if a party member is a BNet or WoW friend
+    for groupindex = 1, 4 do
+        local nameOfPartyMember = UnitName("party"..groupindex)
+        if (nameOfPartyMember) then
+            if Contains(FriendListTable, nameOfPartyMember) then
+                return true
+            end
+        end
+    end
+    
+    return false
+end
+
+function APR.party:GetFriendsList()
```

Comment:
after looking at your code in detail, I've optimised it a bit
- I find for loops more readable than while loops 
- you're making a useless API call (twice)
in general, in programming, when you try to get an object from an function or an API put  this object into a variable before checking something with it like an IF (I don't always do it xD but it's more optimised)


```lua
function APR.party:GetFriendsList()
    local friendsOnlineWoW = C_FriendList.GetNumFriends()
    local _, friendsOnlineBNet = BNGetNumFriends()

    local FriendListTable = {}

    -- add WoW friend
    for i = 1, friendsOnlineWoW do
        local friendInfo = C_FriendList.GetFriendInfoByIndex(i)
        if friendInfo and friendInfo.name then
            tinsert(FriendListTable, friendInfo.name)
        end
    end

    -- add BNet friend
    for i = 1, friendsOnlineBNet do
        local bnetFriendInfo = C_BattleNet.GetFriendAccountInfo(i)
        if bnetFriendInfo and bnetFriendInfo.gameAccountInfo and bnetFriendInfo.gameAccountInfo.characterName then
            tinsert(FriendListTable, bnetFriendInfo.gameAccountInfo.characterName)
        end
    end

    return FriendListTable
end
```

Reply:
I wasnt sure if lua will cause a exception because of nil
Will test the code and then update it - Thanks alot. Increased my understanding of lua 👍